Please add support for QuickCheck 2.9.x

The test suite constrains `QuickCheck ==2.7.*`, but that restrictions seems unnecessary, i.e. the build works fine if that version range is extended to incude 2.8.\* (and possibly 2.9.x as well, though I didn't test that).

For the sake of being explicit, here is the error that I think @peti is alluding to:

```
rgscott@gearloose:~/.../Hacking/Haskell/vector-0.12.0.0$ cabal build
Building vector-0.12.0.0...
Preprocessing library vector-0.12.0.0...
Preprocessing test suite 'vector-tests-O0' for vector-0.12.0.0...
[6 of 6] Compiling Main             ( tests/Main.hs, dist/build/vector-tests-O0/vector-tests-O0-tmp/Main.o )

tests/Main.hs:4:1: error:
    Failed to load interface for ‘Tests.Vector.UnitTests’
    Use -v to see a list of the files searched for.
```

This should an easy fix. I'll work on it.

--- a/vector.cabal
+++ b/vector.cabal
@@ -50,11 +50,6 @@ Extra-Source-Files:
       tests/LICENSE
       tests/Setup.hs
       tests/Main.hs
-      tests/Boilerplater.hs
-      tests/Utilities.hs
-      tests/Tests/Move.hs
-      tests/Tests/Bundle.hs
-      tests/Tests/Vector.hs
       benchmarks/vector-benchmarks.cabal
       benchmarks/LICENSE
       benchmarks/Setup.hs
@@ -183,6 +178,14 @@ test-suite vector-tests-O0
   Default-Language: Haskell2010
   type: exitcode-stdio-1.0
   Main-Is:  Main.hs
+
+  other-modules: Boilerplater
+                 Tests.Bundle
+                 Tests.Move
+                 Tests.Vector
+                 Tests.Vector.UnitTests
+                 Utilities
+
   hs-source-dirs: tests
   Build-Depends: base >= 4.5 && < 5, template-haskell, vector,
                  random,
@@ -213,6 +216,14 @@ test-suite vector-tests-O2
   Default-Language: Haskell2010
   type: exitcode-stdio-1.0
   Main-Is:  Main.hs
+
+  other-modules: Boilerplater
+                 Tests.Bundle
+                 Tests.Move
+                 Tests.Vector
+                 Tests.Vector.UnitTests
+                 Utilities
+
   hs-source-dirs: tests
   Build-Depends: base >= 4.5 && < 5, template-haskell, vector,
                  random,

